File: backend/lib/hotelflux/domain/consumo.ex
defmodule HotelFlux.Domain.Consumo do
  @moduledoc """
  Entidad de dominio INMUTABLE — Consumo de producto cargado a habitación.
  """

  defstruct [
    :id,
    :reserva_id,
    :producto_id,
    :precio_unitario,
    :total,
    :eliminado_en,
    :inserted_at,
    :updated_at,
    cantidad: 1,
    estado: "pendiente",
    eliminado: false
  ]

  @doc """
  Calcula el total del consumo. FUNCIÓN PURA.
  """
  def calcular_total(precio_unitario, cantidad) do
    Decimal.mult(precio_unitario, cantidad)
  end
end

File: frontend-personal/src/domain/entidades/consumo.ts
// ═══════════════════════════════════════════════════════════
// HotelFlux — Entidad: Consumo
// ═══════════════════════════════════════════════════════════

import type { Producto } from './producto';

export interface Consumo {
  readonly id: string;
  readonly reserva_id: string;
  readonly producto_id: string;
  readonly cantidad: number;
  readonly precio_unitario: string;
  readonly total: string;
  readonly estado: string;
  readonly producto?: Producto;
  readonly inserted_at: string;
}
